test/flow/codeGeneration.js

```diff
@@ -0,0 +1,319 @@
+          episode: ?Episode,
+        };
+
+        export type HeroAndFriendsNamesQuery = {
+          hero: ? {
+            name: string,
+            friends: ?Array< {
+              name: string,
+            } >,
+          },
+        };
+      `);
+    });
+
+    it(`should generate fragmented query operations`, function() {
+      const context = this.compileFromSource(`
+        query HeroAndFriendsNames {
+          hero {
+            name
+            ...HeroFriends
+          }
+        }
+
+        fragment HeroFriends on Character {
+          friends {
+            name
+          }
+        }
+      `);
+
+      const source = generateSource(context);
+
+      expect(source).to.include(stripIndent`
+        /* @flow */
+        //  This file was automatically generated and should not be edited.
+
+        export type HeroAndFriendsNamesQuery = {
+          hero: ?(HeroFriendsFragment & {
+            name: string,
+          }),
+        };
+
+        export type HeroFriendsFragment = {
+          friends: ?Array< {
+            name: string,
+          } >,
+        };
+      `);
+    });
+
+    it(`should generate query operations with inline fragments`, function() {
+      const context = this.compileFromSource(`
+        query HeroAndDetails {
+          hero {
+            name
+            ...HeroDetails
+          }
+        }
+
+        fragment HeroDetails on Character {
+          ... on Droid {
+            primaryFunction
+          }
+          ... on Human {
+            height
+          }
+        }
+      `);
+
+      const source = generateSource(context);
+
+      expect(source).to.include(stripIndent`
+        /* @flow */
+        //  This file was automatically generated and should not be edited.
+
+        export type HeroAndDetailsQuery = {
+          hero: ?(HeroDetailsFragment & {
+            name: string,
+          }),
+        };
+
+        export type HeroDetailsFragment = {
+          primaryFunction: ?string,
+          height: ?number,
+        };
+      `);
+    });
+
+    it(`should generate mutation operations with complex input types`, function() {
+      const context = this.compileFromSource(`
+        mutation ReviewMovie($episode: Episode, $review: ReviewInput) {
+          createReview(episode: $episode, review: $review) {
+            stars
+            commentary
+          }
+        }
+      `);
+
+      const source = generateSource(context);
+
+      expect(source).to.include(stripIndent`
+        /* @flow */
+        //  This file was automatically generated and should not be edited.
+
+        // The episodes in the Star Wars trilogy
+        export type Episode =
+          "NEWHOPE" | // Star Wars Episode IV: A New Hope, released in 1977.
+          "EMPIRE" | // Star Wars Episode V: The Empire Strikes Back, released in 1980.
+          "JEDI"; // Star Wars Episode VI: Return of the Jedi, released in 1983.
+
+
+        export type ReviewInput = {
```

Comment:
Everything is now a type, especially inputs. They are easier to exploit than interfaces and maybe more idiomatic in flow.